### PR TITLE
Allow regular expressions on `ChangeTagAttribute`

### PR DESCRIPTION
**Pull request by ammachado**
## What's changed?
Allow the usage of regular expressions on `ChangeTagAttribute`

## What's your motivation?
Replace namespaces/namespaces URIs on `schemaLocation`.

## Anything in particular you'd like reviewers to focus on?
No

## Anyone you would like to review specifically?
No

## Have you considered any alternatives or workarounds?
No

## Any additional context
No

### Checklist
- [x] I've added unit tests to cover both positive and negative cases
- [x] I've read and applied the [recipe conventions and best practices](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices)
- [x] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

**Comment by timtebeek** [comment]
Thanks for getting this started ; what part of this makes it still a draft? Do you want to try it in use first before a review & merge?

File: rewrite-xml/src/main/java/org/openrewrite/xml/ChangeTagAttribute.java [hunk 1]
```diff
@@ -75,25 +83,31 @@ public Xml visitTag(Xml.Tag tag, ExecutionContext ctx) {
...
+                    }
                 }
 
                 if (newValue == null) {
                     //noinspection DataFlowIssue
                     return null;
                 }
 
-                String changedValue = (oldValue != null) ? attribute.getValueAsString().replace(oldValue, newValue): newValue;
+                String changedValue = (oldValue != null) ? stringValue.replaceAll(oldValue, newValue) : newValue;
```
Comment (timtebeek) [inline rewrite-xml/src/main/java/org/openrewrite/xml/ChangeTagAttribute.java:104]: I wonder if we should have separate the handling here between regex and non-regex code paths. `replace(a,b)` did exact replacements, whereas `replaceAll(a,b)` introduces regex matching. Seeing how `oldValue` can contain any string, that might then change the behavior around special characters like dots.

File: rewrite-xml/src/main/java/org/openrewrite/xml/ChangeTagAttribute.java [hunk 2]
```diff
@@ -75,25 +83,31 @@ public Xml visitTag(Xml.Tag tag, ExecutionContext ctx) {
...
                     return attribute;
                 }
-                if(oldValue != null && !attribute.getValueAsString().startsWith(oldValue)) {
-                    return attribute;
+
+                String stringValue = attribute.getValueAsString();
+                if (oldValue != null) {
+                    if (Boolean.TRUE.equals(regex) && !Pattern.matches(oldValue, stringValue)) {
+                        return attribute;
+                    } else if ((regex == null || Boolean.FALSE.equals(regex)) && !stringValue.startsWith(oldValue)) {
```
Comment (timtebeek) [inline rewrite-xml/src/main/java/org/openrewrite/xml/ChangeTagAttribute.java:94]:
```suggestion
                    }
                    if ((regex == null || Boolean.FALSE.equals(regex)) && !stringValue.startsWith(oldValue)) {
```

**Review by timtebeek — COMMENTED** [review]
Minor comment that I think we should be able to work through quickly. Thanks for getting this started!

File: rewrite-xml/src/main/java/org/openrewrite/xml/ChangeTagAttribute.java (hunk 1 above)
Comment (ammachado) [reply rewrite-xml/src/main/java/org/openrewrite/xml/ChangeTagAttribute.java:104]: You're right, I just remembered https://github.com/openrewrite/rewrite-logging-frameworks/pull/120. 😅

**Review by timtebeek — APPROVED** [review]
Perfect, thanks a lot!

**Pull request merged**